public/apps/account/tenant-switch-panel.tsx

```diff
@@ -0,0 +1,251 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { CoreStart } from 'kibana/public';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiRadioGroup,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import {
+  comboBoxOptionToString,
+  stringToComboBoxOption,
+} from '../configuration/utils/combo-box-utils';
+import { selectTenant } from '../configuration/utils/tenant-utils';
+import { fetchAccountInfo } from './utils';
+
+interface TenantSwitchPanelProps {
+  coreStart: CoreStart;
+  username: string;
+  handleClose: () => void;
+}
+
+const GLOBAL_TENANT_KEY_NAME = 'global_tenant';
+const GLOBAL_TENANT_RADIO_ID = 'global';
+const PRIVATE_TENANT_RADIO_ID = 'private';
+const CUSTOM_TENANT_RADIO_ID = 'custom';
+
+export function TenantSwitchPanel(props: TenantSwitchPanelProps) {
+  const [tenants, setTenants] = useState<{ [key: string]: boolean }>({});
```

Comment:
Never saw a case that server returns `{tenantName: false}`, and in this we also didn't check if false. In this case I would suggest just use an simple array or Set.

Reply:
"in this we also didn't check if false". I am using the boolean to check "the user is not granted the access". I think this way, it is robust to handle multiple cases. (key is not included in the return, or key is there but value is false)

Reply:
I see, I said because I saw following code didn't check the boolean:

~~~
  const customTenantOptions = _.keys(tenants).filter((tenant) => {
    return tenant !== GLOBAL_TENANT_KEY_NAME && tenant !== props.username;
  });
~~~

Reply:
Get your point now. Let me add the boolean value check.

Reply:
Actually I don't think need boolean in every rendering. Just filtering out false before setting state should make things easy.

Reply:
Agree on the simplified state. Updated.